semantic-core/src/Core/Parser.hs

```diff
@@ -1,29 +1,38 @@
-{-# LANGUAGE FlexibleContexts, TypeOperators #-}
+{-# LANGUAGE DeriveFunctor, DerivingStrategies, FlexibleContexts, GeneralizedNewtypeDeriving, TypeOperators #-}
 module Core.Parser
   ( core
   , lit
   , expr
   , record
   , comp
-  , lvalue
   ) where
 
 -- Consult @doc/grammar.md@ for an EBNF grammar.
 
 import           Control.Applicative
 import           Control.Effect.Carrier
+import           Control.Monad
 import           Core.Core ((:<-) (..), Core)
 import qualified Core.Core as Core
 import           Core.Name
 import qualified Data.Char as Char
 import           Data.Foldable (foldl')
 import           Data.String
+import           Text.Parser.LookAhead (LookAheadParsing)
 import qualified Text.Parser.Token as Token
 import qualified Text.Parser.Token.Highlight as Highlight
+import qualified Text.Parser.Token.Style as Style
 import           Text.Trifecta hiding (ident)
 
 -- * Identifier styles and derived parsers
 
+newtype CoreParser m a = CoreParser { runCoreParser :: m a }
+  deriving stock Functor
+  deriving newtype (Alternative, Applicative, CharParsing, DeltaParsing, Errable, LookAheadParsing, Monad, MonadPlus, Parsing)
```

Comment:
We don’t need to derive `Functor` stock. Let’s drop `DeriveFunctor` & `DerivingStrategies` and just derive the lot of this with GND.